contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3705,13 +3727,30 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 				{
 					StringInfoData query;
 					RoleSpec   *spec;
+					RoleSpec   *rolspec;
+
 					check_alter_server_stmt(grant_role);
 					spec = (RoleSpec *) linitial(grant_role->grantee_roles);
+					rolspec = (RoleSpec *) linitial(grant_role->granted_roles);
 					initStringInfo(&query);
-					if (grant_role->is_grant)
-						appendStringInfo(&query, "ALTER ROLE dummy WITH createrole createdb; ");
+					
+					/* If sysadmin, provide attribute for role and database priv */
+					if (strlen(rolspec->rolename) == 8 && strncmp(rolspec->rolename, BABELFISH_SYSADMIN, 8) == 0)
+					{
+						if (grant_role->is_grant)
+							appendStringInfo(&query, "ALTER ROLE dummy WITH createrole createdb; ");
+						else
+							appendStringInfo(&query, "ALTER ROLE dummy WITH nocreaterole nocreatedb; ");
+					}
+
+					/* Otherwise, provide attribute for role priv */
 					else
```

Comment:
we have blocked alter server role for any other roles other than sysadmin and securityadmin in the parsing phase itself so this else condition which you are taking about will actually never hit.

Reply:
will modify this check in dbcreator but there also will not add any else condition because that according to me is irrelevant since it will never be the case. On next PR we will just add additional else check to handle another case.

Reply:
@anju15bharti ^